### PR TITLE
Make DOI assignment configurable

### PR DESCRIPTION
**Pull request by emanueldima**

### CODE REVIEW

File: b2share/modules/records/serializers/schemas/datacite.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ class IdentifierSchema(Schema):
 
     def get_doi(self, pids):
         p = [p['value'] for p in pids if p['type'] == 'DOI_RESERVED']
-        return str(p[0])
+        return str(p[0]) if p else ""
```
Comment (nharraud) [inline b2share/modules/records/serializers/schemas/datacite.py]: IMHO `None` would be more appropriate than `""`.

Comment (emanueldima) [inline b2share/modules/records/serializers/schemas/datacite.py]: I tried, the serializer crashes...

Comment (nharraud) [inline b2share/modules/records/serializers/schemas/datacite.py]: I have [found this](https://guidelines.openaire.eu/en/latest/data/use_of_datacite.html#what-s-different)
```
OpenAIRE accepts other persistent identifier schemes and not only a DOI in 1.1 identifierType (M).
```
Maybe we could use ePIC PIDs with PID type "Handle".
See also [the list of PID types accepted by OpenAIRE](https://guidelines.openaire.eu/en/latest/data/field_identifier.html#d-identifiertype)

Comment (emanueldima) [inline b2share/modules/records/serializers/schemas/datacite.py]: Fine, but this particular serializer is for datacite, where DOIs are mandatory.

**Pull request merged**